tuist/tuist
# Default to staticLibrary in product type mapping if automatic product type exists

Resolves <https://github.com/tuist/tuist/issues/6557>
/claim #6557

### Short description 📝

When SPM package has `automatic` and `dynamic library` types, tuist should pick `staticFramework` product type by default. This PR adds code that checks if `products` contains `automatic` library type, if so, choose `staticFramework`.

### How to test the changes locally 🧐

1. Create new tuist project
2. Add Alamofire (5.9.1) dependency to main target
3. See Alamofire is a Dynamic Library before fix
4. Apply PR's changes
5. See Alamofire is a Static Library after fix

### Contributor checklist ✅


### Reviewer checklist ✅
